database/database.go

```diff
@@ -94,3 +94,21 @@ type Database interface {
 	io.Closer
 	health.Checker
 }
+
+// BlockDatabase defines the interface for storing and retrieving blockchain blocks.
+type BlockDatabase interface {
+	// WriteBlock inserts a block into the store at the given height.
+	WriteBlock(height uint64, block []byte) error
+
+	// ReadBlock retrieves a block by its height.
+	ReadBlock(height uint64) ([]byte, error)
```

Comment:
1. Should we use `Put/Get/Has` instead of `Write/Read/HasBlock`? 
2. Do we not support deletion?